src/internal/task/etcd_service.go

```diff
@@ -63,6 +66,61 @@ func (es *etcdService) TaskCount(ctx context.Context, namespace string) (tasks i
 	return nTasks, nClaims, nil
 }
 
+func (es *etcdService) ListTaskF(ctx context.Context, namespace string, f func(string, *Task, bool) error) error {
+	etcdCols := newNamespaceEtcd(es.etcdClient, es.etcdPrefix, namespace)
+	var taskData Task
+	var claim Claim
+	etcdCols.taskCol.ReadOnly(ctx).List(&taskData, col.DefaultOptions(), func(key string) error {
```

Comment:
We are missing a potential list error here, we can just return here to collect it.

Reply:
sorry, don't know how I missed that a second time